subcommands/keys/tuf_utils.go

```diff
@@ -212,44 +212,81 @@ func GetOfflineCreds(credsFile string) (OfflineCreds, error) {
 	return files, nil
 }
 
-func FindTufSigner(keyid, pubkey string, creds OfflineCreds) (*TufSigner, error) {
-	pubkey = strings.TrimSpace(pubkey)
-	for k, v := range creds {
-		if strings.HasSuffix(k, ".pub") {
-			tk := client.AtsKey{}
-			if err := json.Unmarshal(v, &tk); err != nil {
-				return nil, fmt.Errorf("Unable to parse JSON for %s: %w", k, err)
-			}
-			if strings.TrimSpace(tk.KeyValue.Public) == pubkey {
-				pkname := strings.Replace(k, ".pub", ".sec", 1)
-				pkbytes := creds[pkname]
-				tk = client.AtsKey{}
-				if err := json.Unmarshal(pkbytes, &tk); err != nil {
-					return nil, fmt.Errorf("Unable to parse JSON for %s: %w", pkname, err)
-				}
-				keyType, err := parseTufKeyType(tk.KeyType)
-				if err != nil {
-					return nil, fmt.Errorf("Unsupported key type for %s: %s", pkname, tk.KeyType)
-				}
-				pk, err := keyType.ParseKey(tk.KeyValue.Private)
-				if err != nil {
-					return nil, fmt.Errorf("Unable to parse key value for %s: %w", pkname, err)
-				}
-				return &TufSigner{
-					Id:   keyid,
-					Type: keyType,
-					Key:  pk,
-				}, nil
-			}
+func FindOneTufSigner(root *client.AtsTufRoot, creds OfflineCreds, keyids []string) (signer TufSigner, err error) {
+	var signers []TufSigner
+	if signers, err = findTufSigners(root, creds, keyids); err == nil {
+		if len(signers) == 0 {
+			err = fmt.Errorf("Found no active signing key for: %v.", keyids)
+		} else if len(signers) > 1 {
+			err = fmt.Errorf(`Found more than one active signing key for: %v.
+This is an unsupported and insecure way to store private keys.
+Please, provide a keys file which contains a single active signing key.`, keyids)
+		} else {
+			signer = signers[0]
 		}
 	}
-	return nil, fmt.Errorf("Can not find private key for: %s", keyid)
+	return
 }
 
-func findTufRootSigner(root *client.AtsTufRoot, creds OfflineCreds) (*TufSigner, error) {
-	kid := root.Signed.Roles["root"].KeyIDs[0]
-	pub := root.Signed.Keys[kid].KeyValue.Public
-	return FindTufSigner(kid, pub, creds)
+func findTufSigners(root *client.AtsTufRoot, creds OfflineCreds, keyids []string) ([]TufSigner, error) {
```

Comment:
At first glance it looks like passing both `client.AtsTufRoot` and `keyids` and then checking if all keys in `keyids` are present in the root is redundant taking into account how `findTufSigners` is used.

`FindOneTufSigner(curCiRoot, creds, curCiRoot.Signed.Roles["root"].KeyIDs) -> findTufSigners(root, creds, keyids)`.  Effectively, `keyids` refers to the `root`.

Maybe `findTufSigners` is used in different way in some other code places.

Reply:
It is redundant in some cases, and not redundant in other cases.
For example, for TUF targets we need to exclude the online signing key from the list.

Reply:
ok